OMCompiler/Compiler/NBackEnd/Modules/2_Pre/NBAlias.mo

```diff
@@ -644,64 +668,541 @@ protected
 
       else algorithm
         // there is no constant binding -> all others will be replaced by one variable
-        alias_vars := chooseVariableToKeep(list(BVariable.getVarPointer(cr) for cr in set.simple_variables));
+        (alias_vars, collector) := chooseVariableToKeep(list(BVariable.getVarPointer(cr) for cr in set.simple_variables), var_to_keep);
         vars := VariablePointers.fromList(alias_vars);
         eqs := EquationPointers.fromList(set.simple_equations);
         // causalize the system
         (_, comps) := Causalize.simple(vars, eqs);
+        if Flags.isSet(Flags.DEBUG_ALIAS) then
+          print(StringUtil.headline_3("Variable to keep (values of attributes before replacements):") + BVariable.pointerToString(Pointer.access(var_to_keep))+"\n\n");
+        end if;
         // create replacements from strong components
         Replacements.simple(comps, replacements);
+        var_lst := VariablePointers.toList(vars);
+        if Flags.isSet(Flags.DEBUG_ALIAS) then
+          print(StringUtil.headline_4("Attribute collector (before replacements): ") + collector.toString(collector) + "\n");
+        end if;
+        for var in var_lst loop
+          rhs := UnorderedMap.getSafe(BVariable.getVarName(var), replacements, sourceInfo());
+          eq := Equation.makeAssignment(BVariable.toExpression(var), rhs, Pointer.create(0), "TMP", Iterator.EMPTY(), EquationAttributes.default(EquationKind.UNKNOWN, false));
```

Comment:
please create a ` constant String PRE_STR         = "PRE";` like string constant for `TMP` at the start of `NBEquation.mo`